File: drquine/Python/Colleen.py
#This program will print its own source when run
def main():
    #This program will print its own source when run
    a = 0
    a = '#This program will print its own source when run\ndef main():\n    #This program will print its own source when run\n    a = 0\n    a = %r\n    print(a%%a, end="")\n\nif __name__ == "__main__":\n    main()'
    print(a%a, end="")

if __name__ == "__main__":
    main()

File: drquine/C/Colleen.c
#include <stdio.h>
/*
    This program will print its own source when run
*/
void colleen()
{
    char *a = "#include <stdio.h>%2$c/*%2$c    This program will print its own source when run%2$c*/%2$cvoid colleen()%2$c{%2$c    char *a = %3$c%1$s%3$c;%2$c    printf(a, a, 10, 34);%2$c    /*%2$c        This program will print its own source when run%2$c    */%2$c}%2$c%2$cint main()%2$c{%2$c    colleen();%2$c}";
    printf(a, a, 10, 34);
    /*
        This program will print its own source when run
    */
}

int main()
{
    colleen();
}